odoo/fields.py

```diff
@@ -1861,32 +1864,49 @@ def _get_attrs(self, model_class, name):
     _description_strip_classes = property(attrgetter('strip_classes'))
 
     def convert_to_column(self, value, record, values=None, validate=True):
-        if value is None or value is False:
-            return None
-        if self.sanitize:
-            return html_sanitize(
-                value, silent=True,
-                sanitize_tags=self.sanitize_tags,
-                sanitize_attributes=self.sanitize_attributes,
-                sanitize_style=self.sanitize_style,
-                sanitize_form=self.sanitize_form,
-                strip_style=self.strip_style,
-                strip_classes=self.strip_classes)
-        return value
+        return self._convert(value, record, True)
 
     def convert_to_cache(self, value, record, validate=True):
+        return self._convert(value, record, validate)
+
+    def _convert(self, value, record, validate):
         if value is None or value is False:
             return None
-        if validate and self.sanitize:
-            return html_sanitize(
-                value, silent=True,
-                sanitize_tags=self.sanitize_tags,
-                sanitize_attributes=self.sanitize_attributes,
-                sanitize_style=self.sanitize_style,
-                sanitize_form=self.sanitize_form,
-                strip_style=self.strip_style,
-                strip_classes=self.strip_classes)
-        return value
+
+        if not validate or not self.sanitize:
+            return value
+
+        sanitize_vals = {
+            'silent': True,
+            'sanitize_tags': self.sanitize_tags,
+            'sanitize_attributes': self.sanitize_attributes,
+            'sanitize_style': self.sanitize_style,
+            'sanitize_form': self.sanitize_form,
+            'strip_style': self.strip_style,
+            'strip_classes': self.strip_classes
+        }
+
+        if isinstance(self.sanitize, list):
+            if record.user_has_groups(','.join(self.sanitize)):
+                return value
+
+            original_value = record[self.name]
+            if original_value:
+                initial_value_sanitized = html_sanitize(original_value, **sanitize_vals)
+
+                def get_parsed(val):
+                    return etree.tostring(html.fromstring(val))
+
+                # could have been emptied by the sanitizer
+                if not initial_value_sanitized \
+                   or not get_parsed(original_value) == get_parsed(initial_value_sanitized):
```

Comment:
if (not initial_value_sanitized
        or get_parsed(original_value) != get_parsed(initial_value_sanitized)):

> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.

> Use `is not` operator rather than `not ... is`. While both expressions are functionally identical, the former is more readable and preferred.

<https://peps.python.org/pep-0008/>

Reply:
:+1: 
I always hesitate on this one. Thanks.
I went for the same style as the multi line conditions in this file
```py
                if (
                    not initial_value_sanitized
                    or get_parsed(original_value) != get_parsed(initial_value_sanitized)
                ):
```